packages/convertkit/src/hooks/use-convertkit-form.tsx

```diff
@@ -0,0 +1,50 @@
+import {useFormik} from 'formik'
+import * as Yup from 'yup'
+import axios from 'axios'
+import {ConvertkitSubscriber} from '../types'
+import {
+  CONVERTKIT_SIGNUP_FORM,
+  CONVERTKIT_SUBSCRIBE_API_URL,
+} from '@skillrecordings/config'
+
+export function useConvertkitForm({
```

Comment:
encapsulating the state and business logic of the component in this hook, which effectively makes it a simplified wrapper around formik's `useFormik` but implies a contract of required values and simplifies the options

Reply:
love it. shouldn't this be `.ts` instead of `.tsx` since it doesn't return a component?